src/main/java/org/zalando/nakadi/service/subscription/SubscriptionRebalancer.java

```diff
@@ -28,15 +77,15 @@
                 activeSessionIds.stream().mapToInt(activeSessionWeights::get).toArray());
 
         // Stage 1. Select partitions that are not assigned to any EXISTING session.
-        final Set<Partition> toRebalance = Stream.of(currentPartitions)
+        final List<Partition> toRebalance = Stream.of(currentPartitions)
```

Comment:
Why did you change this?

Reply:
Before this change the rabalance could provide different output for the same input which made it difficult to write tests. And I think it's good in general to have a predictable result. I asked @antban - there was no intention to do randomization here so he blessed me to change it to List.

Reply:
Ok, I agree that it is good to have predictable test results.